software/contrib/harmonic_lfos.py

```diff
@@ -52,29 +72,20 @@ def reset(self):
         """Reset all LFOs to zero volts, maintaining their divisions"""
         self.degree = 0
 
-    def change_mode(self):
-        """Change the mode that controls wave shape"""
-        self.modes[self.selected_lfo] = (self.modes[self.selected_lfo] + 1) % self.MODES_COUNT
-        self.save_state()
-
     def get_delay_increment_value(self):
         """Calculate the wait time between degrees"""
         delay = (0.1 - (k1.read_position(100, 1) / 1000)) + (ain.read_voltage(1) / 100)
         return delay, round((((1 / delay) - 10) / 1) + 1)
 
-    def increment_selection(self):
-        """Move the selection to the next LFO"""
-        self.selected_lfo = (self.selected_lfo + 1) % 6
-        self.selected_lfo_start_value = self.get_clock_division()
-
     def save_state(self):
         """Save the current set of divisions to file"""
-        if self.last_saved() < 5000:
+        if self.last_saved() < 3000:
```

Comment:
`[Question]`

Why change this duration?

Reply:
This was entirely for user experience purposes. I found that I could consistently make a change and power off in less than 5 seconds. This resulted in confusing behavior because some changes were saved and some weren’t.

I found that reducing this timeout to 3 seconds “felt” to act as expected. This is likely because it took less than 5 seconds and more than 3 to make a change and find the power switch.

That being said, this is triggering a save almost twice as often. If the concern is the impact on performance, i’m happy to revert. Bad performance is a far worse UX than the occasional unsaved setting.

Reply:
Rather than having a fixed interval for saving, would it make more sense to overhaul this so that it just saves whenever there's been a change? Inside the main loop and/or interrupts you set a flag that the user has changed the set of divisions. Then at the bottom of the loop you check the state of the flag, and save if it's been set to true. Otherwise you don't save.

Reply:
I like that idea. I’ll take a crack at it, but I think I’ll revert this change for now and submit the save behavior as a separate PR

Reply:
That's fair.  I'm a fan of the "save on-demand when-needed" approach; flag that something was changed, and let the main loop handle saving it. But I agree, that can be split out into a separate merge request later.